### PR TITLE
FIT-360 Apply the new shared library for smoke testing to CALS API

### PR DESCRIPTION
**Pull request by mariamcws**
### JIRA Issue Link
- [FIT-360 Apply the new shared library for smoke testing to CALS API](https://osi-cwds.atlassian.net/browse/FIT-360)

### Technical Description

Added Jenkinsfile for build steps to deploy to Pre-Integration and Integration and to execute Smoke Tests against the deployed project.

There are separate existing Jenkinsfiles for pr build and master build which may be refactored later so they are all part of this Jenkinsfile.

### Tests
- [ ] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [x] I have NOT included tests 

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import groovy.transform.Field
+@Library('jenkins-pipeline-utils') _
+
+GITHUB_CREDENTIALS_ID = '433ac100-b3c2-4519-b4d6-207c029a103b'
```
Comment (AVandana-OSI) [inline Jenkinsfile]: Add def and Field annotation to it, it make it consistent

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  stage('Smoke Tests') {
+    if (environment == 'preint') {
+      def gradlePropsText = """
+        cals.api.url=https://calsapi.preint.cwds.io/
+        perry.url=https://web.preint.cwds.io/
+        login.form.target.url=https://web.preint.cwds.io/perry/login
+      """
+       writeFile file: "gradle.properties", text: gradlePropsText
+       def buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'clean integrationTest'
+       publishHTML([allowMissing: true, alwaysLinkToLastBuild: true, keepAll: true, reportDir: 'build/reports/tests/integrationTest', reportFiles: 'index.html', reportName: 'Pre Integration Test Report', reportTitles: ''])
```
Comment (AVandana-OSI) [inline Jenkinsfile]: It says Integration test and in else Smoke test? do we need to publish the report

Comment (mariamcws) [reply Jenkinsfile]: Yes, for now it is running functional/integration test for preint and smoke/system-information test for integration that gets results written to the specified directories.
After we work out token login we may change to run functional/integration test for both

File: Jenkinsfile (hunk 1 above)
Comment (mariamcws) [reply Jenkinsfile]: changed

**Pull request merged**